drivers/sensor/bmi160/bmi160_trigger.c

```diff
@@ -183,17 +234,93 @@ static int bmi160_trigger_anym_set(struct device *dev,
 	return 0;
 }
 
+static int bmi160_trigger_tap_set(struct device *dev,
+				  sensor_trigger_handler_t handler,
+				  bool is_single)
+{
+	struct bmi160_device_data *bmi160 = dev->driver_data;
+	u8_t rval = 0U;
+	u8_t mask;
+
+	if (is_single) {
+		bmi160->handler_tap = handler;
+		mask = BMI160_INT_S_TAP_EN;
+	} else {
+		bmi160->handler_dtap = handler;
+		mask = BMI160_INT_D_TAP_EN;
+	}
+
+	if (handler) {
+		rval = is_single ? BMI160_INT_S_TAP_EN : BMI160_INT_D_TAP_EN;
+	}
+
+	if (bmi160_reg_update(dev, BMI160_REG_INT_EN0, mask, rval) < 0) {
```

Comment:
In general, it is better to propagate error codes (`r = call(); if (r < 0) return r;`)